chain/client/src/adapter.rs

```diff
@@ -138,7 +138,10 @@ pub enum ProcessTxResponse {
 
 #[derive(actix::Message, Debug, PartialEq, Eq)]
 #[rtype(result = "()")]
-pub struct ChunkStateWitnessMessage(pub ChunkStateWitness);
+pub struct ChunkStateWitnessMessage {
+    pub witness: ChunkStateWitness,
+    pub peer_id: PeerId,
```

Comment:
Wait, why are we passing the peer_id with the message?? Shouldn't this be derive from the call to `epoch_manager.get_chunk_producer_info`?

I can see that in the `verify_chunk_state_witness_signature` we are checking whether we correct chunk producer has signed the state witness or not. If the signature matches and yet the state witness is incorrect, we should be banning this chunk_producer.

I still don't see the point of peer_id as anyone sending a malicious message can fake it. @Longarithm @robin-near

Reply:
Ok, taking a more detailed look, seems like we are banning the peer that sent us this message.

But in case the state witness is incorrectly produced by the chunk proposer, why should the peer be banned for that?

Reply:
I'm not familiar with network layer, but seemingly that's the only data which it provides. It's not clear how to get PeerId for AccountId, so I was assuming that chunk state witnesses are sent directly from chunk producers, as it happens now.